lib/Runtime/Library/JavascriptPromise.cpp

```diff
@@ -267,6 +267,132 @@ namespace Js
         return promiseCapability->GetPromise();
     }
 
+    Var JavascriptPromise::EntryAny(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+        ARGUMENTS(args, callInfo);
+        Assert(!(callInfo.Flags & CallFlags_New));
+
+        ScriptContext* scriptContext = function->GetScriptContext();
+
+        AUTO_TAG_NATIVE_LIBRARY_ENTRY(function, callInfo, _u("Promise.any"));
+
+        // 1. Let C be the this value.
+        Var constructor = args[0];
+
+        // 2. If Type(C) is not Object, throw a TypeError exception.
+        if (!JavascriptOperators::IsObject(constructor))
+        {
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedObject, _u("Promise.any"));
+        }
+
+        JavascriptLibrary* library = scriptContext->GetLibrary();
+        Var iterable;
+
+        if (args.Info.Count > 1)
+        {
+            iterable = args[1];
+        }
+        else
+        {
+            iterable = library->GetUndefined();
+        }
+
+        // 3. Let promiseCapability be NewPromiseCapability(C).
+        JavascriptPromiseCapability* promiseCapability = NewPromiseCapability(constructor, scriptContext);
+
+        RecyclableObject* constructorObject = VarTo<RecyclableObject>(constructor);
+
+        JavascriptExceptionObject* exception = nullptr;
+        try
+        {
+            // 4. Let iterator be GetIterator(iterable).
+            RecyclableObject* iteratorRecord = JavascriptOperators::GetIterator(iterable, scriptContext);
+
+            // Assert(!JavascriptOperators::IsConstructor(constructorObject));
+
+            JavascriptArray* values = library->CreateArray(0);
+
+            JavascriptPromiseAllResolveElementFunctionRemainingElementsWrapper* remainingElementsWrapper = RecyclerNewStructZ(scriptContext->GetRecycler(), JavascriptPromiseAllResolveElementFunctionRemainingElementsWrapper);
```

Comment:
Why does the RemainingElements need a wrapper? Looking through this I don't understand why it's not just a bare int instead of a struct containing an int.